src/routes/(console)/apply-credit/+page.svelte

```diff
@@ -64,136 +55,85 @@
-        const methodList = await sdk.forConsole.billing.listPaymentMethods();
-        const filteredMethods = methodList.paymentMethods.filter((method) => !!method?.last4);
-        methods = { paymentMethods: filteredMethods, total: filteredMethods.length };
-        paymentMethodId =
-            selectedOrg?.paymentMethodId ??
-            methods.paymentMethods.find((method) => !!method?.last4)?.$id ??
-            null;
-    }
+    async function applyCouponCredit(org: Organization) {
+        $isSubmitting = true;
+        let error = null;
 
-    async function handleSubmit() {
-        if (!couponForm.checkValidity()) return;
         try {
-            let org: Organization;
-            // Create new org
-            if (selectedOrgId === newOrgId) {
-                org = await sdk.forConsole.billing.createOrganization(
-                    newOrgId,
-                    name,
-                    billingPlan,
-                    paymentMethodId
-                );
-            }
-            // Upgrade existing org
-            else if (selectedOrg?.billingPlan !== billingPlan) {
-                org = await sdk.forConsole.billing.updatePlan(
-                    selectedOrg.$id,
-                    billingPlan,
-                    paymentMethodId,
-                    null
-                );
-            }
-            // Existing pro org
-            else {
-                org = selectedOrg;
-            }
+            await sdk.forConsole.billing.addCredit(org.$id, couponData.code);
+            trackEvent(Submit.CreditRedeem);
+        } catch (e) {
+            error = e;
+            $isSubmitting = false;
+            trackError(error, Submit.CreditRedeem);
+        }
 
-            // Add coupon
-            if (couponData?.code) {
-                await sdk.forConsole.billing.addCredit(org.$id, couponData.code);
-            }
+        addNotification({
+            type: error ? 'error' : 'success',
+            message: error ? error.message : 'Credit applied successfully'
+        });
 
-            // Add budget
-            if (billingBudget) {
-                await sdk.forConsole.billing.updateBudget(org.$id, billingBudget, [75]);
-            }
+        if (!error) {
+            await goto(`${base}/organization-${selectedOrgId}`);
+        }
+    }
 
-            // Add collaborators
-            if (collaborators?.length) {
-                collaborators.forEach(async (collaborator) => {
-                    await sdk.forConsole.teams.createMembership(
-                        org.$id,
-                        ['owner'],
-                        collaborator,
-                        undefined,
-                        undefined,
-                        `${$page.url.origin}${base}/invite`
-                    );
-                });
+    async function handleSubmit() {
+        if (couponData) {
+            if (
+                selectedOrg?.billingPlan === BillingPlan.PRO &&
```

Comment:
I think it's easier to check `!campaign?.plan || campaign.plan === selectedOrg?.billingPlan`